homeassistant/components/climate/daikin.py

```diff
@@ -0,0 +1,275 @@
+"""
+Support for the Daikin HVAC.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/climate.daikin/
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ClimateDevice,
+    ATTR_OPERATION_MODE,
+    SUPPORT_TARGET_TEMPERATURE, SUPPORT_TARGET_HUMIDITY,
+    SUPPORT_FAN_MODE,
+    SUPPORT_OPERATION_MODE, SUPPORT_SWING_MODE,
+    PLATFORM_SCHEMA)
+from homeassistant.const import (
+    CONF_HOST, CONF_NAME,
+    TEMP_CELSIUS, ATTR_TEMPERATURE)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['pydaikin==0.4']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_TARGET_HUMIDITY |
+                 SUPPORT_FAN_MODE |
+                 SUPPORT_OPERATION_MODE |
+                 SUPPORT_SWING_MODE)
+
+DAIKIN_STATE_TO_HA = {
+    'fan': 'Fan only',
+    'dry': 'Dry',
+    'cool': 'Cool',
+    'hot': 'Heat',
+    'off': 'Off',
+    'auto': 'Auto',
+    'auto-1': 'Auto',
+    'auto-7': 'Auto'
+}
+
+HA_STATE_TO_DAIKIN = {value: key for key, value in DAIKIN_STATE_TO_HA.items()}
+
+# Description of how the platform is defined
+# climate:
+#   platform: daikin
+#   host: 192.168.10.26
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Daikin HVAC platform."""
+    devices = []
+
+    name = config.get(CONF_NAME, None)
+
+    if discovery_info is not None:
+        host = discovery_info['ip']
+        _LOGGER.info("Discovered a Daikin AC %s", host)
+    else:
+        host = config.get(CONF_HOST, None)
```

Comment:
This value will always be present. `None` as a default value does not make sense here, I think.